subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AsmBackedClassGenerator.java

```diff
@@ -397,20 +414,33 @@ public ClassGenerationVisitor builder() {
         private final boolean extensible;
         private final boolean providesOwnDynamicObject;
         private final boolean requiresServicesMethod;
-
-        private ClassBuilderImpl(Class<?> type, boolean decorated, String suffix, int factoryId, boolean extensible, boolean conventionAware, boolean managed, boolean providesOwnDynamicObject, boolean requiresServicesMethod) {
+        private final boolean requiresInstantiator;
+
+        private ClassBuilderImpl(
+            Class<?> type,
+            boolean decorated,
+            String suffix,
+            int factoryId,
+            boolean extensible,
+            boolean conventionAware,
+            boolean managed,
+            boolean providesOwnDynamicObject,
+            boolean requiresServicesMethod,
+            boolean instantiatesNestedObjects
+        ) {
             this.type = type;
             this.factoryId = factoryId;
             this.managed = managed;
-            classGenerator = new AsmClassGenerator(type, suffix);
-            visitor = classGenerator.getVisitor();
-            generatedType = classGenerator.getGeneratedType();
-            superclassType = Type.getType(type);
-            mixInDsl = decorated;
+            this.classGenerator = new AsmClassGenerator(type, suffix);
+            this.visitor = classGenerator.getVisitor();
+            this.generatedType = classGenerator.getGeneratedType();
+            this.superclassType = Type.getType(type);
+            this.mixInDsl = decorated;
             this.extensible = extensible;
             this.conventionAware = conventionAware;
             this.providesOwnDynamicObject = providesOwnDynamicObject;
             this.requiresServicesMethod = requiresServicesMethod;
+            this.requiresInstantiator = instantiatesNestedObjects;
```

Comment:
I think the parameter and the field should be called the same here.

Reply:
Done.